lib/plugins/aws/deployFunction/index.test.js

```diff
@@ -139,5 +139,24 @@ describe('AwsDeployFunction', () => {
         fs.statSync.restore();
       });
     });
+
+    it('should read the provided artifact', () => {
+      sinon.stub(fs, 'readFileSync').returns('some dummy data');
+      sinon.stub(serverless.service, 'getFunction').returns({
+        handler: true,
+        package: {
+          artifact: 'foo.zip',
+        },
+      });
+      sinon.stub(awsDeployFunction.provider, 'request').resolves();
+
+      return awsDeployFunction.deployFunction().then(() => {
+        // const expected = 'foo.zip';
+        // expect(fs.readFileSync.calledWithExactly(expected)).to.be.equal(true);
```

Comment:
This test checks when `package.artifact` is provided, it's read by `fs.readFileSync()`.